WordPress/src/main/java/org/wordpress/android/ui/pages/PageListFragment.kt

```diff
@@ -54,61 +60,78 @@ class PageListFragment : Fragment() {
         }
     }
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        (activity!!.application as WordPress).component()!!.inject(this)
-    }
-
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         return inflater.inflate(R.layout.pages_list_fragment, container, false)
     }
 
-    override fun onDestroyView() {
-        viewModel.stop()
-        super.onDestroyView()
-    }
-
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        initViewModel(savedInstanceState)
+        val siteArgument = if (savedInstanceState == null) {
+            activity?.intent?.getSerializableExtra(WordPress.SITE) as SiteModel
+        } else {
+            savedInstanceState.getSerializable(WordPress.SITE) as SiteModel
+        }
+
+        val key = checkNotNull(arguments?.getString(fragmentKey))
 
-        initRecyclerView(savedInstanceState)
+        ifNotNull(activity, siteArgument) { activity, site ->
```

Comment:
I'm not sure if having non-null checks in this case is the way we want to go. Maybe crashing when we don't have arguments/activity is better than getting into a weird state without data. I know we'll probably never get into this state but this might be a good usecase either for `!!` or for `checkNotNull`.